experimental/cloudstrapper/playbooks/roles/cfn/cfnMagmaEksFargate.json

```diff
@@ -0,0 +1,294 @@
+        "VpcId" : { "Ref" : "vpcMagmaEks" },
+        "CidrBlock" : { "Ref": "paramPriv1Cidr" },
+        "AvailabilityZone" : { "Ref": "paramAvlZone2" },
+        "Tags" : [ { "Key" : "Name", "Value" : { "Ref": "paramPriv1SubnetTagName" }} ]
+       }
+    },
+    "subPrivNet2" : {
+      "Type" : "AWS::EC2::Subnet",
+      "Properties" : {
+        "VpcId" : {  "Ref": "vpcMagmaEks" },
+        "CidrBlock" : { "Ref": "paramPriv2Cidr" },
+        "AvailabilityZone" : { "Ref": "paramAvlZone3" },
+        "Tags" : [ { "Key" : "Name", "Value" : { "Ref": "paramPriv2SubnetTagName" }} ]
+      }
+    },
+    "gwInetBridge" : {
+      "Type" : "AWS::EC2::InternetGateway",
+        "Properties" : {
+          "Tags" : [ {"Key" : "Name", "Value" : { "Ref" : "paramGwTagName"}} ]
+        }
+     },
+     "gwInetAttach" : {
+       "Type" : "AWS::EC2::VPCGatewayAttachment",
+         "Properties" : {
+           "VpcId" : { "Ref" : "vpcMagmaEks" },
+           "InternetGatewayId" : { "Ref" : "gwInetBridge" }
+         }
+    },
+    "eipNatGw" : {
+      "Type" : "AWS::EC2::EIP",
+      "DependsOn": "gwInetAttach",
+      "Properties" : {
+         "Domain" : "vpc"
+      }
+    },
+    "gwNatGw" : {
+      "Type" : "AWS::EC2::NatGateway",
+        "Properties" : {
+          "AllocationId" : { "Fn::GetAtt" : ["eipNatGw", "AllocationId"]},
+          "SubnetId" : { "Ref" : "subBridgeNet"},
+          "Tags" : [ {"Key" : "Name", "Value" : { "Ref" : "paramGwTagName"}} ]
+        },
+        "DependsOn": "subBridgeNet"
+    },
+    "rtBridgeInetGw" : {
+      "Type" : "AWS::EC2::RouteTable",
+      "Properties" : {
+        "VpcId" : { "Ref" : "vpcMagmaEks" },
+        "Tags" : [ { "Key" : "Name", "Value" : "Route Table for Bridge Network for Internet Gateway" } ]
+       }
+    },
+    "rteBridgeInetGw" : {
+      "Type" : "AWS::EC2::Route",
+      "Properties" : {
+        "RouteTableId" : { "Ref" : "rtBridgeInetGw" },
+        "DestinationCidrBlock" : "0.0.0.0/0",
+        "GatewayId" : { "Ref" : "gwInetBridge" }
+       }
+    },
+    "rtBridgeInetGwAssocSubnet" : {
+      "Type" : "AWS::EC2::SubnetRouteTableAssociation",
+      "Properties" : {
+        "SubnetId" : { "Ref" : "subBridgeNet" },
+        "RouteTableId" : { "Ref" : "rtBridgeInetGw" }
+       }
+    },
+    "sgVpc": {
+      "Type" : "AWS::EC2::SecurityGroup",
+      "Properties" : {
+        "GroupDescription" : "Allow all TCP to client host",
+	"GroupName": { "Ref": "paramSecGroupName" },
+        "VpcId" : {"Ref" : "vpcMagmaEks"},
+        "SecurityGroupIngress" : [{
+          "IpProtocol" : "-1",
+          "FromPort" : 0,
+          "ToPort" : 65535,
+          "CidrIp" : "0.0.0.0/0"
+        }],
+        "SecurityGroupEgress" : [{
+          "IpProtocol" : "-1",
+          "FromPort" : 0,
+          "ToPort" : 65535,
+          "CidrIp" : "0.0.0.0/0"
+        }]
+      }
+    },
+    "roleMagmaEksCluster": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
```

Comment:
If NMS/Orch wants to utilize AWS IAM for auth rather than the embedded one comes with the magma, wonder we need to change anything here.

Reply:
The two key use-cases this experimental feature is hoping to address are (a) Can we move away from a static, EC2 based compute backend to a more flexible Fargate backend and reduce the Orc8r cost footprint and (b) Can a Fargate based environment for AGWs and Orc8r provide an alternate dev environment that can be used for scale (over EC2 which has some limitations). When Orc8r moves to using Fargate, it will use the embedded policies that come with Magma.